content/en/platform/corda/4.3/enterprise/event-scheduling.md

```diff
@@ -75,6 +75,15 @@ method and if they do not return `null` then that activity is scheduled based on
 encounter but which aren’t related to yourself will not have any activities scheduled.
 
 
+## Scheduled events flow logic
+
+`SchedulableState` must return a fixed time based only upon data in the state. That is because any node that inserts the state into their vault, i.e. the state isRelevant, will run the code to calculate the scheduled time and put in a timer for the deadline returned (if non-null). This is re-done every time node restarts. When timer point is crossed (i.e. if node misses the precise second it is still going to happen), the flow names is started. It is the responsibility of that flow to consume and modify the state to change or cancel the timer. Thus, logic will always retry after restart.
+
+Typically flow must include some code that prevents all nodes sharing the state from separately racing to modify it as then they will battle it out at notary. A classic mistake is to put in a time relative to `now` which will keep firing, because we retest the deadline just before we run the proposed flow to sanity check that state hasn't been consumed. The ledger is used to make this idempotent and safe. Ultimately only one update to state can win due to notary and that should mean that a new version of state will return `null` so no more timers are scheduled.
+
+Again, if state misses deadline, it will schedule flow immediately. Modified state can also schedule some new flow at some new time too if that is desired.
```

Comment:
```suggestion
Again, if the state misses the deadline, it will schedule a flow immediately. Modified states can also schedule some new flows at some new time if that is desired.
```